lib/srv/authhandlers.go

```diff
@@ -583,10 +582,7 @@ func (a *ahLoginChecker) canLoginWithRBAC(cert *ssh.Certificate, ca types.CertAu
 		return trace.Wrap(err)
 	}
 
-	// we don't need to check the RBAC for the node if they are only allowed to join sessions
-	if osUser == teleport.SSHSessionJoinPrincipal && auth.RoleSupportsModeratedSessions(accessChecker.Roles()) {
-		return nil
-	}
+	// TODO(zmb3): check whether auth.RoleSupportsModeratedSessions is still used
```

Comment:
We will either need to reinstate this special case and augment it with separate MFA checks, or create a separate `canJoinWithRBAC` to separate the joining decisions from the login decisions.